src/components/sequencing/SequenceEditor.svelte

```diff
@@ -0,0 +1,103 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+  import { userSequencesRows } from '../../stores/sequencing';
+  import effects from '../../utilities/effects';
+  import Chip from '../ui/Chip.svelte';
+  import CssGrid from '../ui/CssGrid.svelte';
+  import CssGridGutter from '../ui/CssGridGutter.svelte';
+  import MonacoEditor from '../ui/MonacoEditor.svelte';
+  import Panel from '../ui/Panel.svelte';
+
+  export let readOnly: boolean = false;
+  export let sequenceCommandDictionaryId: number | null = null;
+  export let sequenceDefinition: string = '';
+  export let sequenceName: string = '';
+  export let sequenceSeqJson: string = '';
+  export let title: string = 'Sequence - Definition Editor';
+
+  const dispatch = createEventDispatcher();
+
+  let commandDictionaryTsFiles: TypeScriptFile[] = [];
+  let monaco: Monaco;
+
+  $: effects
+    .getTsFilesCommandDictionary(sequenceCommandDictionaryId)
+    .then(tsFiles => (commandDictionaryTsFiles = tsFiles));
+
+  $: if (monaco !== undefined) {
+    const { languages } = monaco;
+    const { typescript } = languages;
+    const { typescriptDefaults } = typescript;
+    const options = typescriptDefaults.getCompilerOptions();
+    typescriptDefaults.setCompilerOptions({ ...options, lib: ['esnext'], strictNullChecks: true });
+  }
+
+  $: if (monaco !== undefined && commandDictionaryTsFiles !== undefined) {
+    const { languages } = monaco;
+    const { typescript } = languages;
+    const { typescriptDefaults } = typescript;
+    typescriptDefaults.setExtraLibs(commandDictionaryTsFiles);
+  }
+
+  function downloadSeqJson() {
+    const a = document.createElement('a');
+    a.href = URL.createObjectURL(new Blob([sequenceSeqJson], { type: 'application/json' }));
+    a.download = sequenceName;
+    a.click();
+  }
+</script>
+
+<CssGrid bind:rows={$userSequencesRows}>
```

Comment:
Small nit that probably isn't part of this PR (can start a new issue) since it affects other pages like scheduling – i'm finding it really hard to drag the sidebar gutter. Could be a z-index issue but to me it looks like it's too narrow and is missing the drag handle that you can see on other pages like /plan

Reply:
Ah yeah good catch. I have a local note about this. Will make a ticket so we can properly track.